lib/ModuleFilenameHelpers.js

```diff
@@ -138,7 +139,10 @@ ModuleFilenameHelpers.createFilename = (
 		);
 		identifier = memoize(() => requestShortener.shorten(module.identifier()));
 		moduleId = () => chunkGraph.getModuleId(module);
-		absoluteResourcePath = () => module.identifier().split("!").pop();
+		absoluteResourcePath = () =>
+			module instanceof NormalModule
+				? module.resource
+				: module.identifier().split("!").pop();
```

Comment:
Why you add this check?

Reply:
@sokra did: https://github.com/webpack/webpack/pull/14628/commits/4024bb7a2f4c072de85a06cb705e5796aa017c70#diff-8615ef447094606b57dfda96ea402f3cd719db42311729c66776a50cabef1445R145

Reply:
Maybe `module.resource` is only available in `NormalModule`